### PR TITLE
Sign ILVerification.dll

### PR DESCRIPTION
**Pull request by jcouv**
TL;DR: Given the ILVerification ships in a nuget package and as far as I understand the guideline is to strongly name assemblies (even on Core), I'm trying to get ILVerification.dll to be strongly-named.

Background:
Over the break, I've revived the [roslyn PR](https://github.com/dotnet/roslyn/pull/37994) to use ILVerify in the compiler tests (side-by-side with PEVerify).
It's working pretty well, but it hit two issues related to ILVerification.dll not having a strong name:
1. In our correctness build legs, we get an compilation error (`CSC : error CS8002: Referenced assembly 'ILVerification, Version=6.0.0.0, Culture=neutral, PublicKeyToken=null' does not have a strong name. [D:\a\_work\1\s\src\Compilers\Test\Core\Microsoft.CodeAnalysis.Test.Utilities.csproj]`)
2. The validation fails in tests on desktop environment because the desktop runtime won't load the assembly (`System.IO.FileLoadException : Could not load file or assembly 'ILVerification, Version=6.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)`)

Relates to https://github.com/dotnet/roslyn/issues/22872

### CODE REVIEW

**Comment by jcouv** [comment]
@jkotas I've been looking at the [build artifacts](https://dev.azure.com/dnceng/public/_build/results?buildId=1539712&view=artifacts&pathAsName=false&type=publishedArtifacts), but I'm having trouble locating the ILVerification assembly to check that it got signed. Would you know how to find it?

**Comment by jkotas** [comment]
I am not able to find it either. I think the binary is missing in the CI published artifacts.

**Review by jkotas — APPROVED** [review]

**Comment by jcouv** [comment]
Thanks. Can this be merged with a single sign-off?

**Comment by jkotas** [comment]
Yep

**Comment by ericstj** [comment]
@jkotas @jcouv shouldn't we use the open source key for this?  Right now it's using `31bf3856ad364e35` which seems unnecessary for a new assembly.

**Comment by ericstj** [comment]
I would recommend setting `<StrongNameKeyId>Open</StrongNameKeyId>` to use the open key, which is a full key.  We do this for libraries, but it looks like this project isn't using those targets.  https://github.com/dotnet/runtime/blob/72426900ce5777adbd6eb19bafceec261cdd2697/src/libraries/Directory.Build.props#L75

Perhaps we should consider moving that default up.

**Comment by jkotas** [comment]
> Perhaps we should consider moving that default up.

https://github.com/dotnet/runtime/pull/63580
